File: day1/day1.py
f = open('input.txt', 'r')

deltas = [int(line.strip()) for line in f]

f.close()

print('final frequency = {}'.format(sum(deltas)))

freq = 0
seen = {}
while True:
    for delta in deltas:
        freq = freq + delta
        if freq in seen:
            print('repeating frequency = {}'.format(freq))
            exit()
        seen[freq] = True

File: day1/day1.js
const fs = require('fs');

const input = fs.readFileSync('input.txt', 'utf8');

const deltas = input
  .trim()
  .split('\n')
  .map(line => parseInt(line));

const frequency = deltas.reduce((acc, freq) => acc + freq, 0);

console.log(`final frequency = ${frequency}`);

let seen = new Set();
let freq = 0;
while (true) {
  for (let delta of deltas) {
    freq += delta;
    if (seen.has(freq)) {
      console.log(`repeating frequency = ${freq}`);
      process.exit();
    }
    seen.add(freq);
  }
}
